### DISCUSSION TITLE
3 switches on a WEMOS D1-mini

### DISCUSSION THREAD

**Discussion by paulnl** [arendst/Tasmota#15609] [Q&A] [answered]
I have a Wemos D1 mini and I have 3 door sensors connected to it. Now I try the configuration below but it doesn't work.

--CODE --
`SwitchMode1 1`
`SwitchTopic 0`

```
Rule1 on Switch1#state=1 do publish stat/%topic%/DOOR1 ON endon on Switch1#state=0 do Publish stat/%topic%/DOOR1 OFF endon
       on Switch2#state=1 do publish stat/%topic%/DOOR2 ON endon on Switch2#state=0 do Publish stat/%topic%/DOOR2 OFF endon
       on Switch3#state=1 do publish stat/%topic%/DOOR3 ON endon on Switch3#state=0 do Publish stat/%topic%/DOOR3 OFF endon
```

`role1 1`

Only with 1 line so 1 switch it works
Anyone have an idea what I'm doing wrong

**Comment by barbudor** [comment, marked as answer]
Maybe because you seem to have applied the Switchmode configuration to only Switch1? 

Beside you do not give more details on the rest of your configuration such as if any relay is setup.

For using Switches as Digital Input there are other modes that's may be more relevant as they do not need rules (like Switchmode 15) and using SwitchText to name the switches to your liking

**Reply by paulnl** [reply]
Thanks for your reaction, i use readswitches on the door. So i have a open and a closed state.

From the D1 mini i take the state from mqtt

**Reply by sfromis** [reply]
Did you try applying your SwitchMode setting to all 3 switches?
